lib/Syntax/SyntaxFactory.cpp.gyb

```diff
@@ -215,6 +215,10 @@ SyntaxFactory::getUnknownKind(SyntaxKind Kind) {
 %     Result = 'SyntaxKind::UnknownDecl'
 %   elif node.syntax_kind.endswith('Token'):
 %     Result = 'SyntaxKind::UnknownToken'
+%   elif node.syntax_kind.endswith('Type'):
+%     Result = 'SyntaxKind::UnknownType'
+%   elif node.syntax_kind.endswith('Pattern'):
```

Comment:
We should use `SyntaxKind::isTypeKind(Kind)` API et al  here (instead of name based heuristics).
E.g. `SimpleTypeIdentifier` *is* `Type` kind.